src/VirtualMonoRepo/eng/common/templates/jobs/vmr-synchronization.yml

```diff
@@ -0,0 +1,76 @@
+###
+### DO NOT MODIFY THIS FILE!
+###
+### This YAML was auto-generated from VmrSyncJobTemplate
+### To make changes, change the C# definition and rebuild its project
+###
+### These steps synchronize code from product repositories into the VMR (https://github.com/dotnet/dotnet)
+
+parameters:
+- name: targetRef
+  displayName: Target revision in dotnet/installer to synchronize
+  type: string
+  default: $(Build.SourceVersion)
+
+- name: vmrBranch
+  displayName: dotnet/dotnet branch to use
+  type: string
+  default: $(Build.SourceBranchName)
+
+jobs:
+- job: Synchronize
+  displayName: Synchronize dotnet/dotnet
+  pool:
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      name: NetCore-Public
+      demands:
+      - ImageOverride -equals Build.Ubuntu.2004.Amd64.Open
+    ${{ else }}:
+      name: NetCore1ESPool-Internal
+      demands:
+      - ImageOverride -equals Build.Ubuntu.2004.Amd64
+  steps:
+  - checkout: self
+    clean: true
+    displayName: Checkout dotnet/installer
+    fetchDepth: 0
+    path: installer
+
+  - checkout: vmr
+    clean: true
+    displayName: Checkout dotnet/dotnet
+    fetchDepth: 0
+    path: vmr
+    timeoutInMinutes: 30
+
+  - script: |-
+      source ./eng/common/tools.sh
+      InitializeDotNetCli true
+      ./.dotnet/dotnet tool restore
+    displayName: Restore toolset
+    workingDirectory: $(Agent.BuildDirectory)/installer
+
+  - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+    - script: |-
+        cp -r $(Agent.BuildDirectory)/installer $(Agent.TempDirectory)/installer
+      displayName: '[PR Only] Prepare dotnet/installer clone'
```

Comment:
Why is the clone necessary?

Reply:
**What's happening**
The PR is building a commit `SHA1` which is on a `branch1` in some `fork/installer` repo. We are requesting a pull into `vmr-branch` (usually `main` or `release/*`).

**What we want**
In the PR build, we ideally want to test the `VMR/main` **+** `vmr-branch1` (can we apply changes from `branch1` onto the VMR).

**What happens instead**
1. We call `darc vmr update installer:SHA1`
2. The tooling sees the `installer` in `source-mappings.json` and clones `https://github.com/dotnet/installer` (from the mapping)
3. There's no commit like that in the repo so it fails to find it and create the patch

I guess the right solution would be to add an option for overriding the remote from the mapping, but I need to think about it. Will log an issue.

Reply:
https://github.com/dotnet/arcade/issues/11386 (also TODO was added in the YAML)